### PR TITLE
[Settings] Fix regression for time oldformat setting

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR fixes a regression in addon settings for leia when using settings with `type="time"` introduced with the settings rework. The issue is documented here: https://forum.kodi.tv/showthread.php?tid=326633&pid=2687641#pid2687641 

## Motivation and Context
Settings with type `time` use the `HH:MM` format. From the wiki:

`<setting id="record_time" type="time" label="32031" default="13:13"/>`

During the setting rework, the parser for the old settings broke the setting since the provided value is smaller than 8 ( it incorrectly uses HH:MM:SS). Also, when changing the setting and despite dialog numeric does not contemplate seconds the stored value includes 59 seconds by default. Example:

User introduces `12:21` -> kodi stores `12:21:59`.

No idea who to ping so feel free to review or ping someone.

## How Has This Been Tested?
Compiled and runtime tested in OSX with a `time` setting in one of my addons.

## Screenshots (if appropriate):

Master:
alt text

This PR:
alt text

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
jenkins build this please

File: xbmc/XBDateTime.cpp [hunk 1]
```diff
@@ -853,7 +853,7 @@ std::string CDateTime::GetAsDBTime() const
   SYSTEMTIME st;
   GetAsSystemTime(st);
 
-  return StringUtils::Format("%02i:%02i:%02i", st.wHour, st.wMinute, st.wSecond);
+  return StringUtils::Format("%02i:%02i", st.wHour, st.wMinute);
```
Comment (Rechi) [inline xbmc/XBDateTime.cpp]: Why is this change needed?

Comment (enen92) [reply xbmc/XBDateTime.cpp]: this is only used by:
https://github.com/xbmc/xbmc/blob/eb161cdb7c4c97e9f054192c9ea9e8d00839c305/xbmc/settings/SettingDateTime.h#L43

The setValue call expects a string in the format `HH:MM` for settings with type `time`. If you prefer I can leave it as  before and only send a substr when calling setValue.

Comment (Montellese) [reply xbmc/XBDateTime.cpp]: IMO you should add a separate method or an argument to choose whether to return with or without seconds (defaulting to with seconds) for this. Because now there is an inconsistency between `CDateTime::GetAsDbTime()` and `CDatetime::GetAsDBDateTime()` which still returns the time in `HH:MM:SS` format.

File: xbmc/XBDateTime.cpp [hunk 2]
```diff
@@ -1056,15 +1056,16 @@ bool CDateTime::SetFromDBDate(const std::string &date)
...
 bool CDateTime::SetFromDBTime(const std::string &time)
 {
-  if (time.size() < 8)
+  if (time.size() < 5)
     return false;
-  // assumes format:
-  // HH:MM:SS
-  int hour, minute, second;
 
+  int hour, minute, second = 0;
```
Comment (Montellese) [inline xbmc/XBDateTime.cpp]: Use a separate line for each initialization.

**Review by Montellese — CHANGES_REQUESTED** [review]
Please address the comments and then this is good to go.

File: xbmc/XBDateTime.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/XBDateTime.cpp]: Thanks @Montellese, nice to see you back.
Yeap I though about that inconsistency. Will do.

**Comment by enen92** [comment]
@Montellese mind taking another look? Decided to create a new method in utils since default parameter approaches are most of the time rejected.

File: xbmc/utils/TimeUtils.h [hunk 3]
```diff
@@ -19,10 +20,25 @@ int64_t CurrentHostFrequency(void);
 class CTimeUtils
 {
 public:
-  static void UpdateFrameTime(bool flip); ///< update the frame time.  Not threadsafe
-  static unsigned int GetFrameTime(); ///< returns the frame time in MS.  Not threadsafe
+  
+  /*!
```
Comment (Montellese) [inline xbmc/utils/TimeUtils.h]: Maybe extract these cosmetic changes into a separate commit.

**Review by Montellese — APPROVED** [review]
Looks good apart from the minor.

**Comment by enen92** [comment]
Jenkins build this please
